test/reactjs/package.json

```
@@ -0,0 +1,38 @@
+{
+  "name": "reactjs",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@craco/craco": "^6.4.3",
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^13.0.0",
+    "@testing-library/user-event": "^13.2.1",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
+    "react-scripts": "5.0.1",
+    "web-vitals": "^2.1.0"
```

Maybe you should also add
```suggestion
    "web-vitals": "^2.1.0"
    "casbin-core": "file:../../"
```

---

@Zxilly do you know how I can fix `ReferenceError: await is not defined` 😅

---

@Shivansh-yadav13 `top-level-await` only works in esm module.
I write it like this
```js
import React, { useEffect, useState } from "react";
import { MemoryAdapter, Model, newEnforcer } from "casbin-core";

const model = new Model(`
    [request_definition]
    r = sub, obj, act

    [policy_definition]
    p = sub, obj, act

    [policy_effect]
    e = some(where (p.eft == allow))

    [matchers]
    m = r.sub == p.sub && r.obj == p.obj && r.act == p.act
    `);

const adapter = new MemoryAdapter(
  `p, alice, data1, read
p, bob, data2, write`);

const Enforcer = ({ sub, obj, act }) => {
  const [loading, setLoading] = useState(false);
  const [result, setResult] = useState(false);
  const [enforcer, setEnforcer] = useState(null);


  useEffect(() => {
    console.log("Enforcer useEffect");
    newEnforcer(model, adapter).then(e => {
      setEnforcer(e);
    });
  }, []);

  useEffect(() => {
    if (enforcer) {
      setLoading(true);
      enforcer.enforce(sub, obj, act).then((res) => setResult(res)).finally(() => setLoading(false));
    }
  }, [sub, obj, act, enforcer]);
  return (
    <>
      <h1>Casbin</h1>
      <h2>Request ({sub}, {obj}, {act})</h2>
      {
        loading ?
          <h3>Enforcing...</h3>
          :
          <h3 data-testid="enforce-result">{result ? "Allowed" : "Denied"}</h3>
      }
    </>
  );
};

export default Enforcer;
```
Maybe can wrap enforcer initialization in `useCallback()`.
@nodece What's your suggestion?

---

I remember support top-level await.

---

@Zxilly This code looks great to me.

---

> Maybe you should also add

Using  `"casbin-core": "link:../../"` for refer to casbin-core/lib.